config/core/resources/apiserversource.yaml

```diff
@@ -59,114 +59,117 @@ spec:
-                  controllerSelector:
-                    type: object
-                    description: "Controller selector restricting the list of objects to watch by a controlling owner reference of the specified kind."
-                    properties:
-                      apiVersion:
-                        type: string
-                        description: "API version of the controlling owner reference."
-                      kind:
-                        type: string
-                        description: "Kind of the controlling owner reference."
-              type: array
-          required:
-          - resources
-          - sink
-          type: object
-        status:
-          properties:
-            conditions:
-              items:
-                properties:
-                  lastTransitionTime:
-                    type: string
-                  message:
-                    type: string
-                  reason:
-                    type: string
-                  severity:
-                    type: string
-                  status:
-                    type: string
-                  type:
-                    type: string
-                required:
-                - type
-                - status
-                type: object
-              type: array
-            sinkUri:
-              type: string
-          type: object
   versions:
   - name: v1alpha1
     served: true
     storage: true
+    schema:
+      openAPIV3Schema:
+        properties:
+          spec:
+            properties:
+              serviceAccountName:
+                type: string
+                description: "name of the ServiceAccount to use to run the receive adapter. More info: https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/."
+              sink:
+                anyOf:
+                - type: object
+                  description: "the destination that should receive events."
+                  properties:
+                    ref:
+                      type: object
+                      description: "a reference to a Kubernetes object from which to retrieve the target URI."
+                      required:
+                      - apiVersion
+                      - kind
+                      - name
+                      properties:
+                        apiVersion:
+                          type: string
+                          minLength: 1
+                        kind:
+                          type: string
+                          minLength: 1
+                        name:
+                          type: string
+                          minLength: 1
+                    uri:
+                      type: string
+                      description: "the target URI. If ref is provided, this must be relative URI reference."
+                - type: object
+                  description: "DEPRECATED: a reference to a Kubernetes object from which to retrieve the target URI."
```

Comment:
will we remove this after 0.13 cut ?

Reply:
likely we will keep it for v1alpha1 until v1alpha1 drops off